app/src/main/java/org/oppia/android/app/recyclerview/BindableAdapter.kt

```diff
@@ -242,10 +195,17 @@ class BindableAdapter<T : Any> internal constructor(
       )
     }
 
-    companion object {
-      /** Returns a new [SingleTypeBuilder]. */
-      inline fun <reified T : Any> newBuilder(): SingleTypeBuilder<T> {
-        return SingleTypeBuilder(T::class)
+    /**
+     * An injectable factory to create [SingleTypeBuilder]s.
+     *
+     * Note that singleTypeBuilders can only be used in fragment-bounded classes.
+     */
+    class Factory @Inject constructor(val fragment: Fragment) {
```

Comment:
Should be ``private val``--it doesn't need to be used outside this class. Ditto below.

Reply:
I think `inline` modifier doesn't allow this, I am getting this error -
```kotlin
Public-API inline function cannot access non-public-API 'private final val fragment: Fragment defined in org.oppia.android.app.recyclerview.BindableAdapter.SingleTypeBuilder.Factory'
```

Reply:
Odd. I haven't seen that before, but this is also probably the first time we've combined an inject constructor with inline. I guess Kotlin does some interesting codegen for inlines that actually expose this field. Not so great, but I suppose it makes sense.